runtime/doc/lsp.txt

```diff
@@ -1647,13 +1700,17 @@ show_line_diagnostics({opts}, {buf_nr}, {line_nr}, {client_id})
 show_position_diagnostics({opts}, {buf_nr}, {position})
                 Open a floating window with the diagnostics from {position}
 
+                Deprecated in favor of
+                |vim.diagnostic.show_position_diagnostics()|
```

Comment:
1. move the helptags for all of these to `runtime/doc/deprecated.txt`
    - in there, only need to move the new replacement, don't need to have full docs
2. don't generate these docs (delete them from lsp.txt). No need for redundant docs.